File: port/swift-jmap/host-logging.c
// MPL-2.0: https://mozilla.org/MPL/2.0/
// Host-test sink only; HarmonyOS builds use CHarmonyLogging/logging.c and HiLog.
#include "logging.h"
#include <stdio.h>
void thunderbird_hilog_info(const char *category, const char *message) {
    fprintf(stderr, "%s: %s\n", category, message);
}
void thunderbird_hilog_debug(const char *category, const char *message) {
    thunderbird_hilog_info(category, message);
}
void thunderbird_hilog_error(const char *category, const char *message) {
    thunderbird_hilog_info(category, message);
}

File: port/swift-imap/host-logging.swift
// MPL-2.0: https://mozilla.org/MPL/2.0/
// Host-test-only OSLog stand-in; not a HarmonyOS runtime adapter.
public struct Logger: Sendable {
    public init(subsystem: String, category: String) {}
    public func info(_ message: Message) {}
    public func debug(_ message: Message) {}
    public func error(_ message: Message) {}
    public struct Message: ExpressibleByStringLiteral, ExpressibleByStringInterpolation {
        public init(stringLiteral: String) {}
        public init(stringInterpolation: StringInterpolation) {}
        public struct StringInterpolation: StringInterpolationProtocol {
            public init(literalCapacity: Int, interpolationCount: Int) {}
            public mutating func appendLiteral(_ literal: String) {}
            public mutating func appendInterpolation<T>(_ value: T) {}
        }
    }
}
